src/util/sss_krb5.c

```diff
@@ -126,10 +126,12 @@ errno_t select_principal_from_keytab(TALLOC_CTX *mem_ctx,
         kerr = krb5_kt_default(krb_ctx, &keytab);
     }
     if (kerr) {
+        const char *krb5_err_msg = sss_krb5_get_error_message(krb_ctx, kerr);
         DEBUG(SSSDBG_FATAL_FAILURE,
               "Failed to read keytab [%s]: %s\n",
                KEYTAB_CLEAN_NAME,
-               sss_krb5_get_error_message(krb_ctx, kerr));
+               krb5_err_msg);
```

Comment:
You do not handle the case where krb5_err_msg is NULL here as you did at a different place.

Maybe this can be made a macro since the pattern occurs at multiple places? But I'm not sure if it would work with complex format strings.

Reply:
> You do not handle the case where krb5_err_msg is NULL here as you did at a different place.

Thank you. Fixed.